### PR TITLE
Update action to Node 16 and update node modules

### PR DESCRIPTION
**Pull request by markpatterson27**
This PR:

- Updates the action to use Node 16
- Updates dependencies to latest versions
- Updates dev dependencies to latest versions

ToDo

- [ ] update eslint settings
- [x] ~explicitly declare chalk as a dependency (rather than an implicite devDependency)~. Added chalk@4 (See warning on chalk [install instructions](https://github.com/chalk/chalk#install))

### CODE REVIEW

**Comment by stevenbitner** [comment]
Version 12 still showing in [.github/workflows/ci.yml ](.github/workflows/ci.yml). This mismatch causes the CI build to fail.

By the way, thanks for this work. I've pulled your changes into my form at https://github.com/stevenbitner/autograding since I don't think any merges happen on this repo any more.

**Comment by markpatterson27** [comment]
The CI update is in PR #55.

I have a combined branch in my fork with successful workflow runs: https://github.com/markpatterson27/autograding/actions/runs/3310743415

File: package.json [hunk 1]
```diff
@@ -17,26 +17,25 @@
...
-    "nock": "^11.7.2",
-    "prettier": "^1.19.1",
-    "ts-jest": "^25.1.0",
-    "typescript": "^3.7.5"
+    "@types/jest": "^29.2.0",
+    "@types/node": "^18.11.3",
+    "@types/uuid": "^8.3.4",
+    "@typescript-eslint/eslint-plugin": "^5.40.1",
+    "@typescript-eslint/parser": "^5.40.1",
+    "@vercel/ncc": "^0.34.0",
```
Comment (jeffrafter) [inline package.json:31]: Curious why this was added here?

**Review by jeffrafter — APPROVED** [review]
Thank you so much for this change 😍 This looks good to me and I've approved it - but I should note that I am not at GitHub and don't have the rights to merge this change moving forward. Tagging @JessRudder who is awesome and might have the power.

File: package.json (hunk 1 above)
Comment (markpatterson27) [reply package.json:31]: You might need to ask the original author :wink: The modules were there from the initial commit. All I did was update them to newer versions.

**Comment by markpatterson27** [comment]
There's still a few changes that should be made:

- eslint rules should be updated, but I'm not sure what the recommended GitHub coding styles are.
- Chalk should be added as an explicite dependency. There are some [breaking changes](https://github.com/chalk/chalk/releases/tag/v5.0.0) going to v5 of Chalk though which I haven't had the bandwidth to look at properly.

**Comment by DerekHarter** [comment]
I suspect this PR would also fix the Issue #71 I just opened as it updates the action/core version.

**Comment by markpatterson27** [comment]
CI workflow run against node 16.x (see PR #55) for these recent commits is in my fork: https://github.com/markpatterson27/autograding/actions/runs/4794051331

**Comment by octosteve** [comment]
Confirmed tests running with node version 16 on fork.

**Pull request merged**